### PR TITLE
[SPARK-29167][SQL] Make Metrics of Analyzer/Optimizer use Scientific counting  human readable

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
Current metrics of Analyzer/Optimizer use Scientific counting is not human readable.
image

change it to a human readable format

### Why are the changes needed?
Make metrics more human readable

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Manuel Test
image

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala [hunk 1]
```diff
@@ -85,10 +85,11 @@ case class QueryExecutionMetering() {
       s"$ruleName $runtimeValue $numRunValue"
     }.mkString("\n", "\n", "")
 
+    val format = new java.text.DecimalFormat("#,##0.#########")
     s"""
        |=== Metrics of Analyzer/Optimizer Rules ===
        |Total number of runs: $totalNumRuns
-       |Total time: ${totalTime / NANOS_PER_SECOND.toDouble} seconds
+       |Total time: ${format.format(totalTime / NANOS_PER_SECOND.toDouble)} seconds
```
Comment (maropu) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: Just use `String.format`?

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: > Just use `String.format`?

`String.format("%.9f, args)` can't handle  extra 0 in the end. 
Such as `String.format("%.9f, (100/ 1000000000D))` = `0.0000000100`

Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: How about this? https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/ui/UIUtils.scala#L50

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: > How about this? https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/ui/UIUtils.scala#L50

`UIUtils.formatDuration(totalTime / NANOS_PER_MICROS)`
Use this method will lose  precision， if you don's mind, it's ok.

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: @maropu  Change to `String.format`

**Comment by dongjoon-hyun** [comment]
Sorry, but why `Scientific counting` is not human-readable? For me, it's also good and readable.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala [hunk 2]
```diff
@@ -21,7 +21,7 @@ import scala.collection.JavaConverters._
 
 import com.google.common.util.concurrent.AtomicLongMap
 
-import org.apache.spark.sql.catalyst.util.DateTimeUtils.NANOS_PER_SECOND
+import org.apache.spark.sql.catalyst.util.DateTimeUtils.{NANOS_PER_SECOND}
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: We don't use `{}` when we import one.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala [hunk 3]
```diff
@@ -85,10 +85,12 @@ case class QueryExecutionMetering() {
       s"$ruleName $runtimeValue $numRunValue"
     }.mkString("\n", "\n", "")
 
+    val totalTimeInJavaDouble =
+      (totalTime / NANOS_PER_SECOND.toDouble).asInstanceOf[java.lang.Double]
     s"""
        |=== Metrics of Analyzer/Optimizer Rules ===
        |Total number of runs: $totalNumRuns
-       |Total time: ${totalTime / NANOS_PER_SECOND.toDouble} seconds
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: How about?
```scala
Total time: ${"%.9f".format(totalTime / NANOS_PER_SECOND.toDouble)} seconds
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala (hunk 2 above)
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: Please revert this change.

**Comment by AngersZhuuuu** [comment]
> Sorry, but why `Scientific counting` is not human-readable? For me, it's also good and readable.

For the normal user , it's a little not intuitive.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala (hunk 2 above)
Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: Done

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala (hunk 3 above)
Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: > How about?
> 
> ```scala
> Total time: ${"%.9f".format(totalTime / NANOS_PER_SECOND.toDouble)} seconds
> ```

Good idea, String.format() need to use java Double type.

**Comment by dongjoon-hyun** [comment]
I'm a normal user, @AngersZhuuuu .
> For the normal user , it's a little not intuitive.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala (hunk 3 above)
Comment (adrian-wang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: maybe we don't need 9 digits after `.`?

Comment (AngersZhuuuu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/QueryExecutionMetering.scala]: > maybe we don't need 9 digits after `.`?

If don't care precision, this [advice](https://github.com/apache/spark/pull/25844#discussion_r325989089) is good.

**Comment by wangyum** [comment]
ok to test

**Comment by wangyum** [comment]
retest this please
